### ISSUE TITLE
uBO WE on Firefox installs differently depending on cookieBehavior setting

### ISSUE THREAD

**Issue by mazesy** [gorhill/uBlock#2993]
Not an issue considering all runs fine, only a surprise.

I'm running Waterfox 55.0.2 on Windows 7 x64

All uBO installs were performed cleanly. uBO is WE 1.14.8 from  https://github.com/gorhill/uBlock/releases

**Install 1**: `network.cookie.cookieBehavior` = 1 (cookies : allow same host)
uBO installs correctly, 
settings located in [Profile]\\browser-extension-data\uBlock0@raymondhill.net\storage.js
data located in [Profile]\storage\default\moz-extension+++fca92df8-70e5-4395-8edf-92abb95618e8

**Install 2**: `network.cookie.cookieBehavior` = 2 (cookies : all disallowed)
uBO installs correctly,
settings **and** data are **both** located in [Profile]\\browser-extension-data\uBlock0@raymondhill.net\storage.js
there is no [Profile]\storage\default\moz-extension+++fca92df8-70e5-4395-8edf-92abb95618e8 folder

I was stunned. uBO with Install 2 seems to perform correctly, as Install 1 it handled correctly retrieval of my backed up settings and it performs filter lists' updates correctly.

I am stunned but as well happier to have uBO's data not located in my profile's storage, mainly because this storage\default folder is shared with sites' Web data.

Besides sharing this experience I'd wish to know if such an Install (cookies blocked, uBO settings and data all in storage.js) is an oddity, if it is acceptable, if any issues are possible, even if as i wrote all runs perfectly well.

**Comment by gorhill** [comment]
Please do not open issue if there is nothing needs to be fixed -- this is not a forum, this is an issue tracker. The indexedDB issue is a browser issue: <https://bugzilla.mozilla.org/show_bug.cgi?id=1313401>.

**Comment by mazesy** [comment]
I posted because I don't know if there is a potential issue in having uBO's data together with its settings in the same storage.js file. I'm not assuming this oddity is uBO's fault but, given what I experience, I only wish to know if I may continue with this install or if I'd rather re-install with cookies allowed. Perhaps I'd rather wait for a true issue to report? I had in mind preventing rather than curing.

**Comment by gorhill** [comment]
> having uBO's data together with its settings in the same storage.js file

How did you conclude the data is saved in `storage.js`?

**Comment by mazesy** [comment]
I didn't conclude, I wasn't in an analysis process, I simply noticed facts, as above described. A 20+MB  [Profile]\browser-extension-data\uBlock0@raymondhill.net\storage.js with no uBO data in the [Profile]\storage\default\moz-extension+++fca92df8-70e5-4395-8edf-92abb95618e8 folder which wasn't even created.

Starting Firefox 55 all seems to turn around the cookies behavior policy. Opened, all can be done by Web sites as well as by extensions, closed and all is blocked. In fact what is applicable to Web data is comparable to what is applied to extensions data. You know, I know, we know it's not uBO's problem but that of the coming FF57's architecture which has serious flaws in the very way data storage is concerned, it's been badly thought, there is a mistake somewhere and I'm not sure it's technical when my opinion is that it is constitutive of the devs' very policy of what they wish : data mixing. If so the concept is unhealthy.

Anyway I've switched back to uBO legacy (Firefox) 1.48.0 and will wait to see what is corrected or not with FF 56 and especially 57.

I'm aware this is not a uBO problem, if I mentioned my experience it is only to be sure that having the install as performed with cookies forbidden is or is not a problem, be it in perspective.
